misk-redis-testing/src/main/kotlin/misk/redis/testing/FakeRedis.kt

```diff
@@ -402,4 +403,12 @@ class FakeRedis @Inject constructor(
   override fun close() {
     // no op
   }
+
+  override fun subscribe(jedisPubSub: JedisPubSub, channel: String) {
+    throw NotImplementedError("Fake client not implemented for this operation")
```

Comment:
nit: I wonder if faking this would be as simple as mutableMapOf<String, MutableList<JedisPubSub>> and listeners[channel].forEach { it.onMessage(message) }

Reply:
the problem is for specifically the `unsubscribe` method